internal/introspection/inbound.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package introspection
+
+// IntrospectableInbound extends the Inbound interface.
+type IntrospectableInbound interface {
+	Introspect() InboundStatus
+}
+
+// InboundStatus is a collection of basics info about an Inbound.
+type InboundStatus struct {
+	Transport string
+	Endpoint  string
+	State     string
+}
```

Comment:
Seems fine to me, especially since it isn't exported.

As a future concern, I'm not convinced by this structured response - we're already encoding arbitrary transport-related stuff into these strings (see the Redis inbound's `Endpoint`, which includes the queue name). For now, totally fine.